### PR TITLE
[Translation] [russian] -  algorithms-fun

### PR DESCRIPTION
**Pull request by alexshd**

### CODE REVIEW

File: _content/tour/rus/algorithms-fun.article [hunk 1]
```diff
@@ -86,46 +83,41 @@ length quantities that they translate to:
...
     Octet   8	7	6	5	4	3	2	1
     #1      0	1	1	0	1	0	1	1
     #2      1	0	0	1	1	0	1	0
 
-Finally we concatenate them, most significant octet first, into
+Наконец, мы объединяем их, старший октет, в
 
 Encoded: 10011010 01101011  ToHex: 0x9A 0x6B
 
-*Extra* *Resources:*
```
Comment (rumyantseva) [inline _content/tour/rus/algorithms-fun.article:122]: Is there a particular reason why this section is excluded from the translation?

Comment (alexshd) [reply _content/tour/rus/algorithms-fun.article:122]: no, no reason at all

File: _content/tour/rus/algorithms-fun.article [hunk 2]
```diff
@@ -42,35 +40,34 @@ leave if all chairs are full.
 
 .play algorithms/fun/barber.go
 
-* Frequency
+* Периодичность
```
Comment (rumyantseva) [inline _content/tour/rus/algorithms-fun.article:43]:
```suggestion
* Частота
```

Feel free to decline, but to me "частота" sounds easier to understand than "периодичность" :)
I didn't mark it when I reviewed it the first time, but now after some addition thinking, I still think "частота" is easier.

File: _content/tour/rus/algorithms-fun.article [hunk 3]
```diff
@@ -42,35 +40,34 @@ leave if all chairs are full.
...
 
-This sample programs shows you how to implement a function that can find
-the frequency of a given rune that is used in a specified sentence.
+Этот пример программы показывает, как реализовать функцию, которая может найти
+периодичность использования данной руны в заданном предложении.
```
Comment (rumyantseva) [inline _content/tour/rus/algorithms-fun.article:46]:
```suggestion
частоту использования данной руны в заданном предложении.
```

File: _content/tour/rus/algorithms-fun.article [hunk 4]
```diff
@@ -42,35 +40,34 @@ leave if all chairs are full.
...
 
-- Sequential: A linear algorithm to perform a rune count.
-- Concurrent: A concurrent algorithm to perform a rune count.
+- Последовательный: Линейный алгоритм для выполнения подсчета рун.
+- Конкурентность: Алгоритм конкурентности для выполнения подсчета рун.
```
Comment (rumyantseva) [inline _content/tour/rus/algorithms-fun.article:49]:
```suggestion
- Конкурентный: Конкурентный алгоритм для выполнения подсчета рун.
```

Just to keep the same style: either "последовательный -- конкруентный" or "последовательность -- конкурентность".

File: _content/tour/rus/algorithms-fun.article [hunk 5]
```diff
@@ -42,35 +40,34 @@ leave if all chairs are full.
...
 
 .play algorithms/fun/freq_sequential.go
 .play algorithms/fun/freq_concurrent.go
 
-*  Variable Length Quantity encoding/decoding.
+* Кодирование/декодирование измеряемой длинны переменной.
```
Comment (rumyantseva) [inline _content/tour/rus/algorithms-fun.article:54]:
```suggestion
* Кодирование/декодирование величин переменной длины.
```

File: _content/tour/rus/algorithms-fun.article [hunk 6]
```diff
@@ -42,35 +40,34 @@ leave if all chairs are full.
...
 
-This sample program showcases how Go can be leveraged to implement variable length quantity encoding/decoding.
+Данный пример программы демонстрирует, как можно использовать Go для реализации кодирования/декодирования измеряемой длинны переменной.
```
Comment (rumyantseva) [inline _content/tour/rus/algorithms-fun.article:56]:
```suggestion
Данный пример программы демонстрирует, как можно использовать Go для реализации кодирования/декодирования величин переменной длины.
```

File: _content/tour/rus/algorithms-fun.article [hunk 7]
```diff
@@ -42,35 +40,34 @@ leave if all chairs are full.
...
+- См. подробнее на [[https://en.wikipedia.org/wiki/Variable-length_code][https://en.wikipedia.org/wiki/Variable-length_code]].
 
-In short, the goal of this encoding is to save encode integer values in
-a way that would save bytes. Only the first 7 bits of each byte is significant
-(right-justified; sort of like an ASCII byte). So, if you have a 32-bit value,
-you have to unpack it into a series of 7-bit bytes. Of course, you will have
-a variable number of bytes depending upon your integer. To indicate which
-is the last byte of the series, you leave bit #7 clear. In all of the
-preceding bytes, you set bit #7.
+Вкратце, цель этого кодирования - сохранить кодировку целых значений таким образом.
```
Comment (rumyantseva) [inline _content/tour/rus/algorithms-fun.article:60]:
```suggestion
Вкратце, цель этого кодирования - сохранить кодировку целых значений
```

File: _content/tour/rus/algorithms-fun.article [hunk 8]
```diff
@@ -42,35 +40,34 @@ leave if all chairs are full.
...
+таким образом, чтобы сэкономить байты. Значимыми являются только первые 7 бит каждого байта
+(выравнивание по правому краю; что-то вроде ASCII-байта). Таким образом, если у вас есть 32-битное значение,
+вам придется упаковать его в серию 7-битных байтов. Конечно, количество байтов будет разным в зависимости от целого числа. Чтобы указать, какой
```
Comment (rumyantseva) [inline _content/tour/rus/algorithms-fun.article:63]:
```suggestion
вам придется упаковать его в серию 7-битных байтов. Конечно, количество байтов будет разным в зависимости от вашего целого. Чтобы указать, какой
```

File: _content/tour/rus/algorithms-fun.article [hunk 9]
```diff
@@ -42,35 +40,34 @@ leave if all chairs are full.
...
+является последним байтом серии, вы оставляете бит #7 чистым. Во всех
```
Comment (rumyantseva) [inline _content/tour/rus/algorithms-fun.article:64]:
```suggestion
байт является последним байтом серии, вы оставляете бит #7 чистым. Во всех
```

File: _content/tour/rus/algorithms-fun.article [hunk 10]
```diff
@@ -42,35 +40,34 @@ leave if all chairs are full.
...
+предыдущих байтах вы устанавливаете бит #7.
 
-So, if an integer is between 0-127, it can be represented as one byte. The
-largest integer allowed is 0FFFFFFF, which translates to 4 bytes variable
-length. Here are examples of delta-times as 32-bit values, and the variable
-length quantities that they translate to:
+Таким образом, если целое число находится в диапазоне 0-127, оно может быть представлено в виде одного байта. 
+Максимальное целое число - 0FFFFFFF, что соответствует 4 байтам    длины переменной.
```
Comment (rumyantseva) [inline _content/tour/rus/algorithms-fun.article:68]:
```suggestion
Максимальное целое число - 0FFFFFFF, что соответствует 4 байтам переменной длины.
```

File: _content/tour/rus/algorithms-fun.article [hunk 11]
```diff
@@ -42,35 +40,34 @@ leave if all chairs are full.
...
+Вот примеры разница-время в виде 32-битных значений и величин переменной
+длины, в которые они преобразуются:
```
Comment (rumyantseva) [inline _content/tour/rus/algorithms-fun.article:70]:
```suggestion
Вот примеры временных интервалов в виде 32-битных значений и величин переменной
длины, в которые они преобразуются:
```

File: _content/tour/rus/algorithms-fun.article [hunk 12]
```diff
@@ -86,46 +83,46 @@ length quantities that they translate to:
...
-integer. It was defined for use in the standard MIDI file format[1] to save
-additional space for a resource constrained system, and is also used in the
-later Extensible Music Format (XMF). A VLQ is essentially a base-128
-representation of an unsigned integer with the addition of the eighth bit
-to mark continuation of bytes. See the example below.
+Код переменной длины (`VLQ`) - это универсальный код, использующий произвольное количество
+двоичных октетов (восьмибитных байтов) для представления произвольно большого
+целого числа. Он был создан для использования в стандартном формате MIDI-файлов[1], чтобы сэкономить
+дополнительное пространство для системы с ограниченными ресурсами, а также используется в более позднем формате
+`Extensible Music Format (XMF)`. VLQ - это, по сути, base-128
```
Comment (rumyantseva) [inline _content/tour/rus/algorithms-fun.article:90]:
```suggestion
`Extensible Music Format (XMF)`. VLQ - это, по сути, 128-ричное
```

File: _content/tour/rus/algorithms-fun.article [hunk 13]
```diff
@@ -86,46 +83,46 @@ length quantities that they translate to:
...
-follows and prepend a 0-bit to the last, signalling the final octet.
+Теперь перед всеми октетами, кроме последнего, мы ставим 1 бит, чтобы указать, что за ним следует октет
+и добавляем 0-бит к последнему, сигнализируя о последнем октете.
 
     Octet   8	7	6	5	4	3	2	1
     #1      0	1	1	0	1	0	1	1
     #2      1	0	0	1	1	0	1	0
 
-Finally we concatenate them, most significant octet first, into
+Наконец, мы объединяем их, старший октет, в
```
Comment (shalimski) [inline _content/tour/rus/algorithms-fun.article:115]:
```suggestion
Наконец, мы объединяем их, начиная со старшего октета, в
```

**Review by shalimski — CHANGES_REQUESTED** [review]

**Review by shalimski — APPROVED** [review]

**Pull request merged**